sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastWithAnsiOnSuite.scala

```diff
@@ -244,7 +244,7 @@ class CastWithAnsiOnSuite extends CastSuiteBase with QueryErrorsBase {
       Decimal("12345678901234567890123456789012345678"))
     checkExceptionInExpression[ArithmeticException](
       cast("123456789012345678901234567890123456789", DecimalType(38, 0)),
-      "Out of decimal type range")
+      "NUMERIC_OUT_OF_SUPPORTED_RANGE")
```

Comment:
Let me just test with `checkExceptionInExpression` for now, since it failed in CI when use `checkError` because `checkError` doesn't support for various test Mode such as "non-codegen mode", "codegen mode" and "unsafe mode" for now.
I think we might need to similar test utils for expression test such as `checkErrorInExpression` something like that.

Reply:
> I think we might need to similar test utils for expression test such as checkErrorInExpression something like that.

Good point. Could you open an JIRA to not forget about this.

Reply:
Yes, just created JIRA here: https://issues.apache.org/jira/browse/SPARK-41208